tests/cpp/test_4in2out.cc

```diff
@@ -0,0 +1,106 @@
+#include <iostream>
+#include <cstdio>
+#include <fstream>
+#include <numeric>
+#include <algorithm>
+#include <functional>
+#include <vector>
+#include <limits>
+#include <stdexcept>
+#include <dlr.h>
+#include <libgen.h>
+
+struct NDArray {
+  std::vector<float> data;
+  std::vector<int64_t> shape;
+  int64_t ndim;
+  int64_t size;
+};
+
+bool is_big_endian();
+std::vector<std::vector<float>> RunInference(DLRModelHandle model);
+
+// Based on hcho3's test harness, this executable 
+// load and test a simple TVM model that sums 2 pairs of tensors 
+// and outputs 2 result tensors.
+//    data1 data2     data3 data4
+//       \  /             \  /
+//      output1          output2
+
+int main(int argc, char** argv) {
```

Comment:
I thin this is not a good way for unit test. We shouldn't expect developers to explicitly provide anything. Let's use gtest directly for different testing cases.

You can find some examples from here:
https://github.com/dmlc/tvm/tree/master/tests/cpp